Civi/Api4/Generic/Traits/ArrayQueryActionTrait.php

```diff
@@ -166,6 +166,8 @@ public static function filterCompare(array $row, array $condition, int $index =
 
       case 'REGEXP':
       case 'NOT REGEXP':
+      case 'REGEXP BINARY':
+      case 'NOT REGEXP BINARY':
         $pattern = '/' . str_replace('/', '\\/', $expected) . '/';
```

Comment:
I believe the pattern should end with `'/i'` for REGEX & NOT REGEX, to match the case-insensitive/case-sensitive behavior of SQL.

Reply:
You could do 
```suggestion
        // Perform case-sensitive matching for BINARY operator, otherwise insensitive
        $i = str_ends_with($operator, 'BINARY') ? '' : 'i';
        $pattern = '/' . str_replace('/', '\\/', $expected) . "/$i";
```

Reply:
Thanks @colemanw - for this review comment, I don't think we need to append /i to the expression, because: REGEXP is not case sensitive, except when used with binary strings. See https://mariadb.com/kb/en/regexp/

Reply:
@agileware-justin this function is in the `ArrayQueryActionTrait` class which is used for *non*-SQL entities. These matching functions are attempting to mimic with PHP what the operator would have done in SQL. It is used on entities that don't use the database, such as the `MockArrayEntity`, so when you update that test you'll see it doesn't work with case-insensitive matches unless you add that bit here.

Reply:
@colemanw thanks - I think to make this behaviour explicit, will also need to add the case-insensitive label to the non-BINARY operators. Rather than the implied behaviour.

It does highlight that the search operator field is not wide enough to show this information without having to click on the field. Expanding the width would solve this problem.

![image](https://github.com/civicrm/civicrm-core/assets/58866555/9ce9457d-ca3c-4302-91e1-37e0a948709a)

Reply:
Expanded width.

![image](https://github.com/civicrm/civicrm-core/assets/58866555/6456dd07-4db6-483a-9aa2-2b48d539ccea)

Reply:
Sounds good @agileware-justin - go ahead & push those updates to this PR and we'll get it across the finish line :)